File: 프로그래머스/연습문제/Lv2/피보나치 수/피보나치 수.py
def solution(n):
    MOD = 1234567
    DP = [0 for i in range(n+1)]
    DP[1] = 1

    for i in range(2, n+1):
        DP[i] = (DP[i-1]+DP[i-2])%MOD

    return DP[n]

File: 프로그래머스/연습문제/Lv2/피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

#define MOD 1234567

int solution(int n) {
    vector<int> DP(n+1, 0);
    DP[1] = 1;
    
    for(int i=2; i<=n; i++)
        DP[i] = (DP[i-1]+DP[i-2])%MOD;
    
    return DP[n];
}
